### PR TITLE
Continuing Nav CSS Refactor

### PR DESCRIPTION
**Pull request by mark-c-woodard**
Continuing along your refactor following some of the concerns you pointed out in your message.
This is still in progress, but I wanted to show it to see if i'm on the right track on the subjective points you brought up.

What I was focusing on here was:
- Move css from the link block to the parent block
- Eliminate duplicate css in navigation block and navigation link
- Remove unused css which appears to be leftovers from the theme.
- Optimize css and reduce very specific css.
- Reduce use of important.
- Improve the css structure.

To that end, I've been trying to keep track of what changes I'm making as I go and here's that list:
**Both Editor and Frontend**
* rename `.dropdown-navigation-toggle` to `kb-nav-dropdown-toggle-btn`
* removing references to `title-dropdown-navigation-toggle`
* remove a menu-item-has-children line because it’s redundant
* rename `vertical-sub-toggle` to `kb-nav-dropdown-toggle-btn`
* combining the nav-link files into the nav files
* remove references to `.submenu` in favor of `.sub-menu`
* consolidating repeat scss selectors
* remove `popup-drawer` references (unused theme code)
* reordering the files some for clarity
* removing references to `menu-label-icon-wrap`
* removing references to `menu-label-icon`
* removing references to `kadence-menu-hidden-title`
* removing references to `.wp-block-kadence-navigation-link__label`
* removing references to `wp-block-kadence-navigation-link__description`
* moving `.link-svg-icon` under `.wp-block-kadence-navigation-link`
* removing `.navigation` wrappers on a lot of styles
  * felt redundant when the style was already wrapped in `.wp-block-kadence-navigation`
* Removing the `li` part from `li.menu` and similar selectors
  * feels redundant and just adds unnecessary specificity 
* generally preferring classnames over tag for readability such as `.menu-item` in favor of `li`
* generally reducing specificity by removing wrapping selectors that are just always present such as `.navigation`
* removing `!important`statements
* removing references to `kb-nav-toggle-sub`
  * this class was always present and just redundant

**Editor**
* reference variables file
* matching changes to `link-highlight-label`
* reducing `.navigation ul.kb-navigation li.wp-block-kadence-navigation-link` to `.wp-block-kadence-navigation-link` to match frontend

### CODE REVIEW

File: includes/blocks/class-kadence-blocks-navigation-link-block.php [hunk 1]
```diff
@@ -326,17 +337,23 @@ public function sized_dynamic_styles( $css, $attributes, $unique_id, $size = 'De
...
-		$css->add_property( 'color', $css->render_color( $sized_attributes['linkColor'] ), $sized_attributes['linkColor'] );
+		//no added specificty needed for these variables
+		//these variable will slot into selectors found in the static stylesheet.
+		$css->set_selector( '.kb-nav-link-' . $unique_id );
+		$css->add_property( '--kb-nav-dropdown-link-color', $css->render_color( $sized_attributes['linkColorDropdown'] ), $sized_attributes['linkColorDropdown'] );		
+		$css->add_property( '--kb-nav-dropdown-link-color-hover', $css->render_color( $sized_attributes['linkColorDropdownHover'] ), $sized_attributes['linkColorDropdownHover'] );
+		$css->add_property( '--kb-nav-dropdown-link-color-active', $css->render_color( $sized_attributes['linkColorDropdownActive']), $sized_attributes['linkColorDropdownActive'] );
+		$css->add_property( '--kb-nav-dropdown-link-color-active-ancestor', $css->render_color( $sized_attributes['linkColorDropdownActive']), $sized_attributes['parentActive'] && $sized_attributes['linkColorDropdownActive'] );
```
Comment (kadencewp) [inline includes/blocks/class-kadence-blocks-navigation-link-block.php:346]: I'm getting warnings: Undefined array key "parentActive"

File: includes/blocks/class-kadence-blocks-navigation-link-block.php [hunk 2]
```diff
@@ -326,17 +337,23 @@ public function sized_dynamic_styles( $css, $attributes, $unique_id, $size = 'De
...
+
+		//no bleed variables (extra specific to beat things like dropdown or top level styling)
+		$css->set_selector( '.kb-nav-link-' . $unique_id . ' > .kb-link-wrap.kb-link-wrap.kb-link-wrap.kb-link-wrap' );
+		$css->add_property( '--kb-nav-link-color', $css->render_color( $sized_attributes['linkColor'] ), $sized_attributes['linkColor'] );		
+		$css->add_property( '--kb-nav-link-color-hover', $css->render_color( $sized_attributes['linkColorHover'] ), $sized_attributes['linkColorHover'] );
+		$css->add_property( '--kb-nav-link-color-active', $css->render_color( $sized_attributes['linkColorActive']), $sized_attributes['linkColorActive'] );
+		$css->add_property( '--kb-nav-link-color-active-ancestor', $css->render_color( $sized_attributes['linkColorActive']), $sized_attributes['parentActive'] && $sized_attributes['linkColorActive'] );
```
Comment (kadencewp) [inline includes/blocks/class-kadence-blocks-navigation-link-block.php:353]: I'm getting warnings: Undefined array key "parentActive"

**Review by kadencewp — COMMENTED** [review]
all css varables in the block stylesheet should have a fallback default. For example: 

`width: var(--kb-nav-dropdown-link-width);`

Should instead be:

`width: var(--kb-nav-dropdown-link-width, 200px);`

**Comment by mark-c-woodard** [comment]
The ones without fallbacks are styles that didn't have a default before so I didn't include on in the variable. A lot of the things like widths are like that. I tried not to change existing behavior in these cases.

**Pull request merged**